python/cudf_polars/cudf_polars/utils/config.py

```diff
@@ -108,11 +108,39 @@ class ShuffleMethod(str, enum.Enum):
     RAPIDSMPF = "rapidsmpf"
 
 
+T = TypeVar("T")
+
+
+def _make_default_factory(
```

Comment:
This is probably a big "Nit", but is the typing a bit "off" for `sink_to_directory`?
For `sink_to_directory`, the default type is `None` (rather than `bool`).